src/Assignments/SimpleProblemPage.tsx

```diff
@@ -6,98 +6,305 @@ import ProblemIframe from './ProblemIframe';
         setLoading(true);
         (async () => {
             try {
-                if(_.isNil(params.topicId)) {
+                if (_.isNil(params.topicId)) {
                     console.error('topicId is null');
                     setError('An unexpected error has occurred');
                     return;
                 }
-                const res = await getQuestions({
-                    userId: 'me',
-                    courseTopicContentId: parseInt(params.topicId, 10)
-                });
-                const problems: Array<ProblemObject> = res.data.data.questions;
-                
-                const topic = res.data.data.topic;
-
-                if (topic.studentTopicOverride?.length > 0) {
-                    _.assign(topic, topic.studentTopicOverride[0]);
-                }
-                setTopic(topic);
-
-                if (!_.isEmpty(problems)) {
-                    const problemDictionary = _.chain(problems)
-                        .map(prob => {
-                            const override = prob.studentTopicQuestionOverride?.[0];
-                            if (!_.isNil(override)) {
-                                delete override.id;
-                                _.assign(prob, override);
-                            }
-                            return new ProblemObject(prob);
-                        })
-                        .keyBy('id')
-                        .value();
-                    setProblems(problemDictionary);
-                    setSelectedProblemId(problems[0].id);
-                }
 
+                await fetchProblems(parseInt(params.topicId, 10));
                 setLoading(false);
             } catch (e) {
                 setError(e.message);
                 console.error(e);
                 setLoading(false);
             }
         })();
-    }, [params.topicId]);
+    }, [params.topicId, versionId]);
+
+    const fetchProblems = async (topicId: number) => {
+        const res = await getQuestions({
+            userId: 'me',
+            courseTopicContentId: topicId
+        });
+        const problems: Array<ProblemObject> = res.data.data.questions;
+
+        const topic = res.data.data.topic;
+
+        if (topic.studentTopicOverride?.length > 0) {
+            _.assign(topic, topic.studentTopicOverride[0]);
+        }
+        setTopic(topic);
+
+        if (!_.isEmpty(problems)) {
+            const problemDictionary = _.chain(problems)
+                .map(prob => {
+                    const override = prob.studentTopicQuestionOverride?.[0];
+                    if (!_.isNil(override)) {
+                        delete override.id;
+                        _.assign(prob, override);
+                    }
+                    return new ProblemObject(prob);
+                })
+                .keyBy('id')
+                .value();
+            setProblems(problemDictionary);
+            setSelectedProblemId(_.sortBy(problems, ['problemNumber'])[0].id);
```

Comment:
https://lodash.com/docs/4.17.15#minBy
```suggestion
            setSelectedProblemId(_.minBy(problems, 'problemNumber').id);
```

Reply:
nifty!

Reply:
TS falls apart because it claims ignorance of the fact that 'problemNumber' is a key for the ProblemObject[]

Reply:
🤔